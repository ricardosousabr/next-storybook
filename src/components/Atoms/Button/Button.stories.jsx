import Button from "./index.jsx";

export default {
  title: "Components/Button",
  Component: Button,
  parameters: {
    backgrounds: {
      values: [
        { name: "dark", value: "#000000" },
        { name: "white", value: "#ffffff" },
        { name: "purple", value: "#b220d7" },
      ],
    },
  },
};

export const Primary = (args) => <Button {...args} />;
Primary.args = {
  padding: '',
  borderRadius: '',
  background: 'color',
  href: "",
  children: "",
  type: "submits",
  onClick: () => {
    console.log("Test");
  },
};
