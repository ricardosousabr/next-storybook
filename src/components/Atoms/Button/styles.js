import { styled } from "styled-components";

export const ButtonStyle = styled.button`
  background:  ${(props) => props.background};
  border: none;
  border-radius: ${(props) => props.borderRadius};
  padding: ${(props) => props.padding} ;

`;
