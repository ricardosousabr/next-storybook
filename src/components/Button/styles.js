import { styled } from "styled-components";

export const BoxButton = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100vh;
`;

export const ButtonAdd = styled.button`
  background: aqua;
  border: none;
  width: 100px;
  height: 40px;
  border-radius: 14px;
`;
