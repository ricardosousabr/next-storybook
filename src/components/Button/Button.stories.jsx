import Button from "./index.jsx";

export default {
  title: "Components/Button",
  Component: Button,
};

export const Primary = (args) => <Button {...args}></Button>;
Primary.args = {
  children: "Button",
  type: "submits",
  onClick: () => {
    console.log("Test");
  },
};
