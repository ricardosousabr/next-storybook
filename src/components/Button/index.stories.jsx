import Button from "./index.jsx";

export default {
  title: "Components/Button",
  Component: Button,
  argTypes: {
    variant: {
      options: ["primary", "secondary"],
      control: { type: "radio" },
    },
  },
};

export const HomePage = () => <Button />;
