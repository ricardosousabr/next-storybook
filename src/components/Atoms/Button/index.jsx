import { ButtonStyle } from "./styles";
import PropTypes from "prop-types";

export default function Button({ children, type, onClick, href, background, borderRadius, padding }) {
  const buttonType = {
    type: type,
    onClick: onClick,
  }
  const anchorType = {
    as: "a",
    href: href
  }

  return <ButtonStyle padding={padding} borderRadius={borderRadius} background={background} { ...href ? anchorType : buttonType }>{children}</ButtonStyle>;
}

Button.propTypes = {
  padding: PropTypes.string,
  borderRadius: PropTypes.string,
  background: PropTypes.string,
  href: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.string,
  type: PropTypes.oneOf(["submit", "reset", "button"]).isRequired,
};

Button.defaultProps = {
  children: "click",
  type: "submit"
};