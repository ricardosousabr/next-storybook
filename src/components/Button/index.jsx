import { BoxButton, ButtonAqua, ButtonPurple } from "./styles";
import PropTypes from "prop-types";

export default function Button({ children, type, onClick, href, size }) {
  return (
    <>
      <BoxButton>
        {href ? (
          <ButtonAqua href={href} as="a" type={type}>
            {children}
          </ButtonAqua>
        ) : (
          <ButtonAqua type={type} onClick={onClick}>
            {children}
          </ButtonAqua>
        )}
      </BoxButton>
    </>
  );
}

Button.propTypes = {
  href: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.string,
  size: PropTypes.oneOf(["smail", "medium", "large"]),
  type: PropTypes.oneOf(["submit", "reset", "button"]).isRequired,
};

Button.defaultProps = {
  children: "ttt",
};
