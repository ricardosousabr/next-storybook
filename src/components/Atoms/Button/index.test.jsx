import renderer from "react-test-renderer";
import Button from "./index";

test("render button", () => {
  const tree = renderer.create(<Button padding="10px 20px" borderRadius="12px" background="aqua" type="button" onClick={() => console.log("Test")}>Click</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
