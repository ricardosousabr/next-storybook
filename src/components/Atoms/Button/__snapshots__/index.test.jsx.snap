// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render button 1`] = `
<button
  background="aqua"
  borderRadius="12px"
  className="styles__ButtonStyle-sc-1hhwn61-0 bFpobR"
  onClick={[Function]}
  padding="10px 20px"
  type="button"
>
  Click
</button>
`;
